src/ctl/ctl-sink.c

```diff
@@ -87,50 +89,46 @@ static void sink_handle_get_parameter(struct ctl_sink *s,
 		return cli_vERR(r);
 
 	/* wfd_content_protection */
-	if (rtsp_message_read(m, "{<>}", "wfd_content_protection") >= 0) {
-		r = rtsp_message_append(rep, "{&}",
-					"wfd_content_protection: none");
-		if (r < 0)
-			return cli_vERR(r);
-	}
+	check_and_response_option("wfd_content_protection", "none");
 	/* wfd_video_formats */
-	if (rtsp_message_read(m, "{<>}", "wfd_video_formats") >= 0) {
-		char wfd_video_formats[128];
-		sprintf(wfd_video_formats,
-			"wfd_video_formats: 00 00 03 10 %08x %08x %08x 00 0000 0000 10 none none",
+	char wfd_video_formats[128];
+	sprintf(wfd_video_formats, "wfd_video_formats: 00 00 03 10 %08x %08x %08x 00 0000 0000 10 none none",
 			s->resolutions_cea, s->resolutions_vesa, s->resolutions_hh);
-		r = rtsp_message_append(rep, "{&}", wfd_video_formats);
-		if (r < 0)
-			return cli_vERR(r);
-	}
+//	check_and_response_option("wfd_video_formats", wfd_video_formats);
+	check_and_response_option("wfd_video_formats", "40 00 01 10 0001bdeb 051557ff 00000fff 10 0000 001f 11 0780 0438, 02 10 0001bdeb 155557ff 00000fff 10 0000 001f 11 0780 0438");
 	/* wfd_audio_codecs */
-	if (rtsp_message_read(m, "{<>}", "wfd_audio_codecs") >= 0) {
-		r = rtsp_message_append(rep, "{&}",
-					"wfd_audio_codecs: AAC 00000007 00");
-		if (r < 0)
-			return cli_vERR(r);
-	}
+//	check_and_response_option("wfd_audio_codecs", "wfd_audio_codecs: AAC 00000007 00");
+	check_and_response_option("wfd_audio_codecs", "LPCM 00000003 00, AAC 0000000f 00, AC3 00000007 00");
```

Comment:
This looks ok, anyway maybe not all the audio formats are really supported on sink